src/HumanoIDs.php

```diff
@@ -6,33 +6,42 @@
 
 final class HumanoIDs
 {
+    private static ?HumanoID $zooGenerator = null;
+    private static ?HumanoID $spaceGenerator = null;
+
     public static function zooIdGenerator(
         ?string $separator = '-',
         ?WordFormatOption $format = null
     ): HumanoID {
-        return new HumanoID(
-            json_decode(
-                file_get_contents(__DIR__ . '/../data/zoo-words.json'),
-                true
-            ),
-            null,
-            $separator,
-            $format
-        );
+        if (self::$zooGenerator === null) {
```

Comment:
I'm def a fan of explicit null by default and checking for null equality.
Top notch.